### ISSUE TITLE
testClientSucceedsWithVerificationDisabled fails on master

### ISSUE THREAD

**Issue by dnhatn** [elastic/elasticsearch#38212]
`testClientSucceedsWithVerificationDisabled` fails on FipsJVM. I think we expect this test not to run on FipsJVM but mistakenly use `assertFalse` instead of `assumeFalse`.

CI: 
- https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/216/console

- https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java8fips,nodes=immutable&&linux&&docker/216/console

```
 ./gradlew :modules:reindex:unitTest \
  -Dtests.seed=3868B83F74B6BE50 \
  -Dtests.class=org.elasticsearch.index.reindex.ReindexRestClientSslTests \
  -Dtests.method="testClientSucceedsWithVerificationDisabled" \
  -Dtests.security.manager=true \
  -Dtests.locale=da \
  -Dtests.timezone=Africa/Asmera \
  -Dcompiler.java=12 \
  -Druntime.java=8FIPS \
  -Djavax.net.ssl.keyStorePassword=password \
  -Djavax.net.ssl.trustStorePassword=password
```

**Comment by jkakavas** [comment]
That's correct Nhat, I will fix and update this as it's already weekend for Tim

**Comment by dnhatn** [comment]
Thanks @jkakavas :)

**Comment by talevy** [comment]
opened https://github.com/elastic/elasticsearch/pull/38871 to backport this fix to 6.7
